.github/workflows/openjdk_14_build.yml

```
@@ -0,0 +1,117 @@
+name: openjdk_14_build
+on:
+  pull_request:
+  push:
+    branches: [ main, develop ]
+    tags: [ v* ]
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    container: gcr.io/swirlds-registry/cci-openjdk-ghr:14.0.2
+    env:
+      MAVEN_OPTS: -Xmx5200m
+      LC_ALL: C.UTF-8
+      DEBIAN_FRONTEND: noninteractive
+    timeout-minutes: 60
+    steps:
+      - run: apt-get -y install software-properties-common
+      - run: add-apt-repository ppa:git-core/ppa
+      - run: apt-get -y install git
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - run: mvn --version
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Maven packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
```

Use `actions/setup-java@v2` and it has caching built-in:

```suggestion
      - name: Install JDK
        uses: actions/setup-java@v2
        with:
          cache: 'maven'
          distribution: 'temurin'
          java-version: 14
```